### ISSUE TITLE
Attachments - Retain Original Upload Date

### ISSUE THREAD

**Issue by m-prodan** [bcgov/foi-flow#4200]
* As a user
* I want the original upload date to be retained when I edit a files name or tag 
* so that it preserves more accurately when things were added

**Assumptions & Scope**	
This is based on user feedback we've received regarding the attachment log. When a user renames a file, the upload date changes. The users would like the original upload date to remain static when they make changes to filenames in the attachment log.

**What is IN scope?**
Renaming a file, reclassifying file tag (#3398)

**What is OUT of Scope?**
Replacing a file - this should always update the upload date

**Acceptance Criteria**	

**Scenario 1: Rename File**
* GIVEN a user is on the attachment log
* WHEN they rename a file
* THEN the upload date will remain the same as the original upload

**Scenario 2: Re-Classify File**
* GIVEN a user is on the attachment log
* WHEN they reclassify a tag on the file
* THEN the upload date will remain the same as the original upload

**Dependencies? What is the impact of this dependency? (If so, link dependency in the ticket, make it visible in a team´s backlog)**

Reclassifying file tag (#3398) may need to be done before, or in tandem, with this ticket (as this ticket references this feature)

**Validation Rules? (If yes, list here)**	

**Design**
@xxx - please link the Design here

**Definition of Ready**

1. [ ] Is there a well articulated User Story?
2. [ ] Is there Acceptance Criteria that covers all scenarios (happy/sad paths)?
3. [ ] If there is a user interface, is there a design? 
4. [ ] Does the user story need user research/validation?
5. [ ] Does this User Story needs stakeholder approval?
6. [ ] Design / Solution accepted by Product Owner
7. [ ] Is this user story small enough to be completed in a Sprint? Should it be split?
8. [ ] Are the dependencies known/ understood? (technical, business, regulatory/policy)
9. [ ] Has the story been estimated?

**Definition of Done**

1.    [ ] Passes developer unit tests
2.    [ ] Passes peer code review
3.    [ ] If there's a user interface, passes UX assurance
4.    [ ] Passes QA of Acceptance Criteria with verification in Dev and Test
5.	[ ] Confirm Test cases built and succeeding 
6.	[ ] No regression test failures
7.	[ ] Test coverage acceptable by Product Owner
8.	[ ] Ticket ready to be merged to master or story branch
9.	[ ] Developer to list Config changes/ Update documents and designs
10.	[ ] Can be demoed in Sprint Review 
11.	[ ] Tagged as part of a Release
12.   [ ] Feature flagged if required
13.  [ ] Change Management activities done?

**Comment by m-prodan** [comment]
Hi @milosdes - it might be that we should have had an AC for this, but under the assumptions for this 'Replacing' a file should update the upload date. Users only really have this option if a Ministry user uploads a file as part of a state change (you handled something similar with the re-classify story). Do you think you could fix this quick?

**Comment by m-prodan** [comment]
Per discussion - moving this one back to review, let me know when its been resolved and I'll finish the QA on this one. Thanks!

**Comment by milosdes** [comment]
Chatted with @m-prodan  - 
We currently aren't storing any kind of replaced_at date explicitly, we might want to consider doing that for audit log purposes at some point in the future.
For now, we just want to display the date when the document was replaced on the frontend.

**Comment by milosdes** [comment]
The issue with the date changing when using the "Replace" function for an attachment has been fixed

**Comment by m-prodan** [comment]
Working as expected now - thank you for the fix @milosdes
